dev/com.ibm.ws.cdi.ejb_fat/fat/src/com/ibm/ws/cdi/ejb/tests/NewEjbTimerTest.java

```diff
@@ -0,0 +1,79 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.ejb.tests;
+
+import static componenttest.rules.repeater.EERepeatTests.EEVersion.EE7_FULL;
+import static componenttest.rules.repeater.EERepeatTests.EEVersion.EE9;
+
+import java.io.File;
+import java.util.logging.Logger;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.FileAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.annotation.TestServlets;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.rules.repeater.EERepeatTests;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * Test that library jars inside a war can be implicit bean archives.
+ */
+@RunWith(FATRunner.class)
+public class NewEjbTimerTest extends FATServletClient {
+
+    private static final Logger LOG = Logger.getLogger(NewEjbTimerTest.class.getName());
+
+    public static final String SERVER_NAME = "cdi12EJB32Server";
+    public static final String EJB_TIMER_APP_NAME = "newEjbTimer";
+
+    //not bothering to repeat with EE8 ... the EE9 version is mostly a transformed version of the EE8 code
+    @ClassRule
+    public static RepeatTests r = EERepeatTests.with(SERVER_NAME, EE9, EE7_FULL);
+
+    @Server(SERVER_NAME)
+    @TestServlets({
+                    @TestServlet(servlet = com.ibm.ws.cdi.ejb.apps.newtimer.TestEjbTimerServlet.class, contextRoot = EJB_TIMER_APP_NAME)
+    })
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        WebArchive ejbTimer = ShrinkWrap.create(WebArchive.class, EJB_TIMER_APP_NAME + ".war")
+                                        .addPackage(com.ibm.ws.cdi.ejb.apps.newtimer.SessionScopedBean.class.getPackage())
+                                        .add(new FileAsset(new File("test-applications/" + EJB_TIMER_APP_NAME + ".war/resources/META-INF/permissions.xml")),
+                                             "/META-INF/permissions.xml")
+                                        .add(new FileAsset(new File("test-applications/" + EJB_TIMER_APP_NAME + ".war/resources/WEB-INF/beans.xml")), "/WEB-INF/beans.xml");
```

Comment:
I would prefer these files to be in the `com.ibm.ws.cdi.ejb.apps.newtimer` package, next to the application classes, rather than in `test-applications`.

Reply:
I agree, but didn't want to introduce an inconsistency in the bucket. At some point we'll be moving the lot across.